### ISSUE TITLE
s390 image

### ISSUE THREAD

**Issue by tpimh** [ClangBuiltLinux/boot-utils#11]
As buildroot no longer supports s390, an alternative should be found.

Relevant issue: https://github.com/ClangBuiltLinux/continuous-integration/issues/232

**Comment by nathanchance** [comment]
https://github.com/ClangBuiltLinux/continuous-integration/pull/229#issuecomment-583760615 needs to be dealt with first.

```
$ timeout 15s unbuffer qemu-system-s390x -M s390-ccw-virtio -kernel out/s390x/arch/s390/boot/bzImage -m 512 -display none  -serial mon:stdio
KASLR disabled: CPU has no PRNG
Linux version 5.7.0-rc3 (nathan@ubuntu-s3-xlarge-x86) #1 SMP Tue Apr 28 00:12:23 MST 2020Kernel fault: interruption code 0005 ilc:2
PSW : 0000200180000000 00000000000177ea
      R:0 T:0 IO:0 EX:0 Key:0 M:0 W:0 P:0 AS:0 CC:2 PM:0 RI:0 EA:3
GPRS: 0000000000000008 0000000c00000000 00000003fffffff4 0000000000000002
      0000000000000000 00000000fffffff8 000000000000000c 00000000fffffff0
      00000000fffffffc 0000000000000000 00000000fffffff0 000000000086b5a8
      0000000000000009 0000000000000001 00000000fffffff4 000000000000bce0
```

We should be able to boot all the way until `init`.

I've started messing around with creating a simple `busybox` initramfs though.

**Comment by nathanchance** [comment]
An incoherent brain dump around the busybox initramfs: https://gist.github.com/nathanchance/3c947281c4fe246be45cdd3dcd4b1e8c

cc @nickdesaulniers from https://github.com/ClangBuiltLinux/continuous-integration/issues/232.

It might be nice to reach out to the s390 folks about it, along with asking if they have done any formal testing around clang (since getting it simply building vs. booting is a completely different story).

**Comment by nickdesaulniers** [comment]
@nathanchance was that ever fixed?

**Comment by nathanchance** [comment]
WIP: #26

**Comment by nathanchance** [comment]
Looks like QEMU side fixes are needed for clang s390 kernels to boot?

https://lore.kernel.org/qemu-devel/20210107202135.52379-1-david@redhat.com/

https://lore.kernel.org/qemu-devel/20210107231322.56345-1-david@redhat.com/

I will mess around with those two patches and #26 tomorrow.

**Comment by nickdesaulniers** [comment]
fixed in #28
